apps/papra-server/src/modules/intake-emails/username-drivers/pattern/pattern.intake-email-username-driver.ts

```diff
@@ -0,0 +1,52 @@
+import slugify from '@sindresorhus/slugify';
+import { createError } from '../../../shared/errors/errors';
+import { createLogger } from '../../../shared/logger/logger';
+import { isNil } from '../../../shared/utils';
+import { parseEmailAddress } from '../../intake-emails.models';
+import { defineIntakeEmailUsernameDriverFactory } from '../intake-email-username.models';
+import { PATTERN_INTAKE_EMAIL_ADDRESSES_DRIVER_NAME } from './pattern.intake-email-username-driver.config';
+import { PATTERNS_PLACEHOLDERS } from './pattern.intake-email-username-driver.constants';
+
+export const patternIntakeEmailUsernameDriverFactory = defineIntakeEmailUsernameDriverFactory(({
+  logger = createLogger({ namespace: 'intake-emails.addresses-drivers.pattern' }),
+  config,
+  usersRepository,
+  organizationsRepository,
+}) => {
+  const { pattern } = config.intakeEmails.username.drivers.pattern;
+
+  return {
+    name: PATTERN_INTAKE_EMAIL_ADDRESSES_DRIVER_NAME,
+    generateIntakeEmailUsername: async ({ userId, organizationId }) => {
+      const [{ user }, { organization }] = await Promise.all([
+        usersRepository.getUserById({ userId }),
+        organizationsRepository.getOrganizationById({ organizationId }),
+      ]);
+
+      if (isNil(user) || isNil(organization)) {
+        // Should not really happen, there is a check on the routes handlers
+        throw createError({
+          message: 'User or organization not found',
+          code: 'intake-emails.addresses.user_or_organization_not_found',
+          statusCode: 404,
+        });
+      }
+
+      const { username: userEmailUsername } = parseEmailAddress({ email: user.email });
+
+      const rawUsername = pattern
+        .replaceAll(PATTERNS_PLACEHOLDERS.USER_NAME, user.name ?? '')
+        .replaceAll(PATTERNS_PLACEHOLDERS.USER_ID, user.id)
+        .replaceAll(PATTERNS_PLACEHOLDERS.USER_EMAIL_USERNAME, userEmailUsername)
+        .replaceAll(PATTERNS_PLACEHOLDERS.ORGANIZATION_ID, organization.id)
+        .replaceAll(PATTERNS_PLACEHOLDERS.ORGANIZATION_NAME, organization.name)
```

Comment:
The organization.name can contain spaces and " symbols. Sanitazation is needed or else the name cant be inserted in the database.

Reply:
The db layer auto handle database serialization and sanitazation, no need for an higher level processing